libc-test/build.rs

```diff
@@ -387,6 +399,29 @@ fn main() {
             "KERN_KDENABLE_BG_TRACE" if apple => true,
             "KERN_KDDISABLE_BG_TRACE" if apple => true,
 
+            // uClibc doesn't support a LOT of constants, because its latest release
+            // was before any of them existed
+            "F_GETPIPE_SZ" | "F_SETPIPE_SZ" |
+            "CLOCK_MONOTONIC_RAW" | "CLOCK_REALTIME_COARSE" | "CLOCK_MONOTONIC_COARSE" | "CLOCK_BOOTTIME" | "CLOCK_REALTIME_ALARM" | "CLOCK_BOOTTIME_ALARM" |
+            "SOL_NETBEUI" | "SOL_LLC" | "SOL_DCCP" | "SOL_NETLINK" | "SOL_TICP" | "AF_RDS" | "PF_RDS" |
+            "MSG_FASTOPEN" | "AT_NO_AUTOMOUNT" | "AT_EMPTY_PATH" | "AF_IB" | "AF_MPLS" | "AF_NFC" | "AF_VSOCK" |
+            "PF_IB" | "PF_MPLS" | "PF_NFC" | "PF_VSOCK" |
+            "MSG_COPY" | "EPOLLEXCLUSIVE" |
+            "AIO_CANCELED" | "AIO_NOTCANCELED" | "AIO_ALLDONE" | 
+            "LIO_READ" | "LIO_WRITE" | "LIO_NOP" | "LIO_WAIT" | "LIO_NOWAIT" |
+            "PR_MPX_ENABLE_MANAGEMENT" | "PR_MPX_DISABLE_MANAGEMENT" | "PR_SET_FP_MODE" | "PR_GET_FP_MODE" |
+            "PR_FP_MODE_FR" | "PR_FP_MODE_FRE" | "PR_CAP_AMBIENT" | "PR_CAP_AMBIENT_IS_SET" | "PR_CAP_AMBIENT_RAISE" | "PR_CAP_AMBIENT_LOWER" | "PR_CAP_AMBIENT_CLEAR_ALL" | "O_TMPFILE" | "SOL_TIPC" | "CLONE_NEWCGROUP" | "SO_INCOMING_CPU" | "SO_ATTACH_BPF" | "SO_DETACH_BPF" |
+            // or they weren't implemented yet
+            "LC_PAPER" | "LC_NAME" | "LC_ADDRESS" | "LC_TELEPHONE" | "LC_MEASUREMENT" | "LC_IDENTIFICATION" |
+            "LC_PAPER_MASK" | "LC_NAME_MASK" | "LC_ADDRESS_MASK" | "LC_TELEPHONE_MASK" | "LC_MEASUREMENT_MASK" | "LC_IDENTIFICATION_MASK" | "LC_ALL_MASK" |
+            "MS_DIRSYNC" | "MS_MOVE" | "MS_REC" | "MS_SILENT" | "MS_POSIXACL" | "MS_UNBINDABLE" | "MS_PRIVATE" | "MS_SLAVE" | "MS_SHARED" | "MS_RELATIME" | "MS_KERNMOUNT" | "MS_I_VERSION" | "MS_STRICTATIME" | "MS_ACTIVE" |
+            "QIF_BLIMITS" | "QIF_SPACE" | "QIF_ILIMITS" | "QIF_INODES" | "QIF_BTIME" | "QIF_ITIME" | "QIF_LIMITS" | "QIF_USAGE" | "QIF_TIMES" | "QIF_ALL" | "IP_TRANSPARENT" |
+            "Q_GETFMT" | "Q_GETINFO" | "Q_SETINFO" | "SHM_EXEC" | "RB_SW_SUSPEND" | "RB_KEXEC" | "RUSAGE_THREAD" |
+            // not entirely sure why these don't work...
+            "LC_CTYPE_MASK" | "LC_NUMERIC_MASK" | "LC_TIME_MASK" | "LC_COLLATE_MASK" | "LC_MONETARY_MASK" | "LC_MESSAGES_MASK" |
+            "MADV_MERGEABLE" | "MADV_UNMERGEABLE" | "MADV_HWPOISON" | "IPV6_ADD_MEMBERSHIP" | "IPV6_DROP_MEMBERSHIP" | "IPV6_MULTICAST_LOOP" | "IPV6_V6ONLY" |
+            "MAP_STACK" | "MAP_HUGETLB" | "RTLD_DEEPBIND" |
+            "SOL_IPV6" | "SOL_ICMPV6" | "SOL_TIPC"  if uclibc => true,
```

Comment:
Fixed (most of them)! The couple that are left appear to exist in the uClibc source code, but they're either optional or I'm missing some compiler flags to trigger their inclusion. I'm pretty sure at least the ones related to locale are optional